### ISSUE TITLE
[PRE REVIEW]: Xono2L: an eXtensible, LabVIEW-compatible real-time sONOgraphic data capture and processing TOOL

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1104]
**Submitting author:** @dzhoshkun (<a href="http://orcid.org/0000-0003-3009-4178">Dzhoshkun Ismail Shakir</a>)
**Repository:** <a href="https://github.com/gift-surg/Xono2L" target ="_blank">https://github.com/gift-surg/Xono2L</a>
**Version:** v0.1.0
**Editor:** @Kevin-Mattheus-Moerman
**Reviewer:** Pending

**Author instructions**

Thanks for submitting your paper to JOSS @dzhoshkun. The JOSS editor (shown at the top of this issue) will work with you on this issue to find a reviewer for your submission before creating the main review issue.

@dzhoshkun if you have any suggestions for potential reviewers then please mention them here in this thread. In addition, [this list of people](https://bit.ly/joss-reviewers) have already agreed to review for JOSS and may be suitable for this submission.

**Editor instructions**

The JOSS submission bot @whedon is here to help you find and assign reviewers and start the main review. To find out what @whedon can do for you type:

```
@whedon commands
```

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @yochannah it looks like you're currently assigned as the editor for this paper :tada:

For a list of things I can do to help you, just type:

```
@whedon commands
```

**What happens now?**

This submission is currently in a `pre-review` state which means we are waiting for the assigned editor (@yochannah) to find some reviewers for your submission. This may take anything between a few hours to a couple of weeks. Thanks for your patience :smile_cat:

You can help the editor by looking at [this list of potential reviewers](https://bit.ly/joss-reviewers) to identify individuals who might be able to review your submission (please start at the bottom of the list). Also, feel free to suggest individuals who are not on this list by mentioning their GitHub handles here.

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir<sup>*</sup>, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by arfon** [comment]
:wave: @yochannah - the submitting author suggested you as the handling editor.

**Comment by yochannah** [comment]
Hey! @arfon @dzhoshkun  - I'm not sure if I'm the best person to edit this. I'm not familiar with C++ or Labview, and it's not a scientific domain I'm really familiar with either.

**Comment by arfon** [comment]
OK np @yochannah. @kyleniemeyer can you find someone else to edit this for us?

**Comment by kyleniemeyer** [comment]
Hey @brainstorm @Kevin-Mattheus-Moerman would one of you be able to edit this submission?

**Comment by brainstorm** [comment]
I would love to, but LabVIEW is not OSS but proprietary, closed software. Is there any other means to review this software that does not require to run LabVIEW?

**Comment by dzhoshkun** [comment]
Thanks @yochannah No worries. I wasn't sure about whom to suggest either, but just saw the keyword "API" in your short biography, so this is why I suggested you.

**Comment by Kevin-Mattheus-Moerman** [comment]
I'm contacting some potential reviewers.

**Comment by dzhoshkun** [comment]
Thanks @brainstorm 

The software is conceptualised as an extensible API that builds into a DLL (please note that it still needs the proprietary Sonix SDK for capturing live ultrasound data). We had to make some effort for making it LabVIEW compatible. And therefore we wanted to make it available to the community, such that for instance other research groups with similar setups could benefit from and/or extend its capabilities.

This probably does not answer your question, but I guess my point is that as LabVIEW compatibility is a main selling point, it'd make more sense for it to be reviewed by someone who has access to LabVIEW.

**Comment by dzhoshkun** [comment]
Thanks @Kevin-Mattheus-Moerman

**Comment by dzhoshkun** [comment]
Regarding the PDF compilation error (https://github.com/openjournals/joss-reviews/issues/1104#issuecomment-442479759): it seems to be related to the [fake markdown footnote](https://stackoverflow.com/a/25584585) I've introduced for indicating joint first authorship. Any recommendation how this should/could be achieved in the journal's manuscript format ?

**Comment by jedbrown** [comment]
Try [Pandoc footnotes](https://pandoc.org/MANUAL.html#footnotes)

**Comment by dzhoshkun** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Compile the paper
@whedon generate pdf

🚧 🚧 🚧 Experimental Whedon features 🚧 🚧 🚧

# Compile the paper from a custom git branch
@whedon generate pdf from branch custom-branch-name

```

**Comment by dzhoshkun** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch fix-joint-1-author-fn

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch fix-joint-1-author-fn. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch fix-joint-1-author-fn

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch fix-joint-1-author-fn. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch fix-joint-1-author-fn

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch fix-joint-1-author-fn. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-c44264c87f39/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
Even though I [added quotes around multiple affiliation indices](https://raw.githubusercontent.com/gift-surg/Xono2L/fix-joint-1-author-fn/paper/paper.md) as [shown in the example](https://raw.githubusercontent.com/arfon/fidgit/master/paper/paper.md), the PDF generation still fails with the same error. Any ideas?

**Comment by dzhoshkun** [comment]
@jedbrown Thanks for the hint on padnoc footnotes.

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch fix-joint-1-author-fn

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch fix-joint-1-author-fn. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch fix-joint-1-author-fn

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch fix-joint-1-author-fn. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf from branch this-branch-does-not-exist

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch this-branch-does-not-exist. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1104 with the following error: 

 /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:58:in `block in build_affiliation_string': Problem with affiliations for Dzhoshkun Shakir[, perhaps the affiliations index need quoting? (RuntimeError)
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:57:in `build_affiliation_string'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/author.rb:17:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:109:in `block in parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `each'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:106:in `parse_authors'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon.rb:73:in `initialize'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `new'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/lib/whedon/processor.rb:32:in `set_paper'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:52:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-443822f01563/bin/whedon:113:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by dzhoshkun** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01104/joss.01104/10.21105.joss.01104.pdf)

**Comment by dzhoshkun** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01104/joss.01104/10.21105.joss.01104.pdf)

**Comment by dzhoshkun** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01104/joss.01104/10.21105.joss.01104.pdf)

**Comment by dzhoshkun** [comment]
Apologies for cluttering the thread with the PDF generation commands. I'm giving up on using/faking footnotes for indicating co-first authorship, as this yields undesired artefacts in the generated PDF. Doing it inline instead. @arfon I hope this is acceptable?

**Comment by danielskatz** [comment]
I think the way you've done it is fine.

**Comment by dzhoshkun** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01104/joss.01104/10.21105.joss.01104.pdf)

**Comment by dzhoshkun** [comment]
@FaustinCarter Would you be able to (help) edit this submission?

**Comment by dzhoshkun** [comment]
Adding @tvercaut to this thread as well.

**Comment by kyleniemeyer** [comment]
@Kevin-Mattheus-Moerman it looks like you are planning to edit this submission, so I will go ahead and formalize that 😄

**Comment by kyleniemeyer** [comment]
@whedon assign @Kevin-Mattheus-Moerman as editor

**Comment by whedon** [comment]
OK, the editor is @Kevin-Mattheus-Moerman

**Comment by FaustinCarter** [comment]
@dzhoshkun Unfortunately I no longer have access to LabVIEW so I'd be unable to run it/test it.

**Comment by dzhoshkun** [comment]
@FaustinCarter No problem. Thanks for letting us know.

**Comment by Kevin-Mattheus-Moerman** [comment]
@dzhoshkun I've contacted quite a few people but so far all work on different hardware/software so did not feel they were able to review this work. I'll continue my search but in the mean time, are you able to suggest reviewers?

**Comment by FaustinCarter** [comment]
You might try asking @petebarry1988. Don't know if he'll have time or be willing, but he's got the background.

**Comment by dzhoshkun** [comment]
@Kevin-Mattheus-Moerman I've asked a number of people with a relevant background to review this paper. Will update you if and when I hear back from them.

**Comment by lassoan** [comment]
@dzhoshkun contacted me to review this submission - here are my comments:

It is nice that you took the effort to share this software. Unfortunately, the potential number of users of the software is very small, because very few people would consider using LabView for ultrasound imaging (especially in the open-source community) and the software only supports Ultrasonix machines, which are discontinued (production stopped about a year ago and can no longer be purchased).

I don't think you can find a reviewer who would independently verify "functionality" of the software (according to https://joss.readthedocs.io/en/latest/review_criteria.html) because you need LabView, Ultrasonix, and willingness to spend time with it. (in my case: I don't have LabView and do not have time to spend on this)

**Comment by dzhoshkun** [comment]
@lassoan Thank you very much for your comments and the insight.

**Comment by davidbradway** [comment]
I echo the comments of @lassoan . Though I have access to LabView and an ultrasound research lab, I use Verasonics hardware rather than Ultrasonix. And though I applaud @dzhoshkun sharing your work in a Github repository, I don't think this is the appropriate Journal since it relies on closed-source hardware and software tools.

**Comment by FaustinCarter** [comment]
From a closer look at their code, I don't think anyone actually needs LabVIEW to test it. Their API is accessible from LabVIEW, but it appears that it could be tested with any language that can call methods exposed in a DLL. That loosens the requirements for review slightly to someone who has access to the Sonix SDK, C++ compiler, and knows how to build a library and call functions from a DLL.

From the guidelines: 

> submissions that rely upon a proprietary/closed source language or development environment are acceptable provided that they meet the other submission requirements and that you, the reviewer, are able to install the software & verify the functionality of the submission

So basically (in my opinion) it shouldn't be rejected, but the authors may need to come to peace with the fact that finding a reviewer could take a very long time.

**Comment by labarba** [comment]
👋  @Kevin-Mattheus-Moerman — Have you made a determination whether this submission can be reviewed?

**Comment by Kevin-Mattheus-Moerman** [comment]
@labarba I have been unable to find reviewers for this submission. Perhaps we are forced to reject this issue at the moment as it appears too challenging to find reviewers at this point.

**Comment by labarba** [comment]
@dzhoshkun  — Thank you for submitting your software to JOSS. Unfortunately, we haven’t been able to secure reviewers for your submission, which we (the editors) link to a software’s probability of being useful to others. In this case, a reliance on a particular brand of ultrasound device (apparently discontinued) combined with a LabView-focused solution, make the software difficult to review. It also means it’s probably of narrow use, which puts it somewhat on the fringe of scope for JOSS. At this point, I recommend withdrawing the paper.
Cc. @openjournals/joss-eics

**Comment by dzhoshkun** [comment]
Thank you @labarba @Kevin-Mattheus-Moerman and everyone for all your efforts. Much appreciated. It proved to be really difficult to find reviewers for this work, which has raised admittedly valid concerns about how useful the software is to the community. So I agree that withdrawing this paper is the best course of action at this stage.

**Comment by dzhoshkun** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

```
